MAVProxy/modules/mavproxy_misseditor/__init__.py

```diff
@@ -235,13 +248,35 @@ def child_task(self, q, l, gq, gl):
         self.app.frame.set_event_queue_lock(l)
         self.app.frame.set_gui_event_queue(gq)
         self.app.frame.set_gui_event_queue_lock(gl)
+        self.app.frame.set_close_window_semaphore(cw_sem)
 
+        self.app.SetExitOnFrameDelete(True)
         self.app.frame.Show()
+
+        # start a thread to monitor the "close window" semaphore:
+        watcher_quitsem = multiprocessing.Semaphore()
+        watcher_quitsem.acquire()
+        class CloseWindowSemaphoreWatcher(threading.Thread):
+            def __init__(self, task, sem, quitsem):
+                threading.Thread.__init__(self)
+                self.task = task
+                self.sem = sem
+                self.quitsem = quitsem
+            def run(self):
+                while not self.sem.acquire(False) and not self.quitsem.acquire(False):
+                    time.sleep(0.1)
```

Comment:
do we really need a thread waking up 10 times a second just to monitor window close?